src/main/java/redis/clients/jedis/JedisSentinelPool.java

```diff
@@ -15,30 +17,15 @@
 
 public class JedisSentinelPool extends JedisPoolAbstract {
 
-  /**
-   * @deprecated This will be private in future.
-   */
-  @Deprecated
-  protected static Logger log = LoggerFactory.getLogger(JedisSentinelPool.class);
+  private static final Logger LOG = LoggerFactory.getLogger(JedisSentinelPool.class);
 
   protected final GenericObjectPoolConfig<Jedis> poolConfig;
 
-  protected final int connectionTimeout;
-  protected final int soTimeout;
-  protected final int infiniteSoTimeout;
+  protected final JedisClientConfig masterClientConfig;
 
-  protected final String user;
-  protected final String password;
-  protected final int database;
-  protected final String clientName;
+  protected final JedisClientConfig sentinelClientConfig;
 
-  protected int sentinelConnectionTimeout;
-  protected int sentinelSoTimeout;
-  protected String sentinelUser;
-  protected String sentinelPassword;
-  protected String sentinelClientName;
-
-  protected final Set<MasterListener> masterListeners = new HashSet<>();
+  protected final Collection<MasterListener> masterListeners = new LinkedList<>();
```

Comment:
@sazzad16 why did you move from Set to List?

Reply:
@gkorland I didn't find any logic to put `MasterListener`s in a Set. For example, to put in a HashSet, the respective class should override hashCode and equals methods.

As this is a breaking change, I included it here. Even if we consider that there are codes that use masterListeners, hopefully this won't break many codes as Collection is super class to Set.

Reply:
Doesn't even seem like MasterListener implements equals/hashcode, so this wouldn't have been working as expected anyway.

Why LinkedList specifically though? ArrayList are usually more efficient

Reply:
@mina-asham IMO, the number of sentinels are usually very low, may be 3-5 in most use cases. Java would create 16-size array with the empty constructor of ArrayList. So I though to avoid that.

Only usage of this collection/list is in shutdown and only operation is a for-each loop. There's no special gain for ArrayList in this case.

Reply:
Default size for array list is 10, but I think it's still more efficient from a size point of view in anyway, each LinkedList item is stored in a node that stores 3 pointer (data+next+prev), so if you have 3-5 items that means 9-15 pointers, the array will have 1 pointer per item regardless, so it can be less/more by a few bytes here, so I don't think the size comparison matter much.

The iteration perfomance is usually way better though because the array is contiguous in memory, but again for 3-5 items it wouldn't matter much here.

So I think both are more or less the same for the size of the data, and because ArrayList is the one that's used for most cases and using other ones IMO usually mean there is a specific reason for it, that's why I would opt for ArrayList.